na_ucx_keyres_epchk.patch

```diff
@@ -0,0 +1,123 @@
+diff --git a/src/na/na_ucx.c b/src/na/na_ucx.c
+index 84eb8b0..3289357 100644
+--- a/src/na/na_ucx.c
++++ b/src/na/na_ucx.c
+@@ -1666,6 +1666,9 @@ na_ucp_listener_destroy(ucp_listener_h listener)
+ static void
+ na_ucp_listener_conn_cb(ucp_conn_request_h conn_request, void *arg)
+ {
++
++    char host_string[NI_MAXHOST];
++    char serv_string[NI_MAXSERV];
+     struct na_ucx_class *na_ucx_class = (struct na_ucx_class *) arg;
+     ucp_conn_request_attr_t conn_request_attrs = {
+         .field_mask = UCP_CONN_REQUEST_ATTR_FIELD_CLIENT_ADDR};
+@@ -1683,20 +1686,35 @@ na_ucp_listener_conn_cb(ucp_conn_request_h conn_request, void *arg)
+             UCP_CONN_REQUEST_ATTR_FIELD_CLIENT_ADDR) == 0,
+         error, "conn attributes contain no client addr");
+ 
++    int rc = getnameinfo((const struct sockaddr *) &conn_request_attrs.client_address,
++                         sizeof(struct sockaddr), host_string,
++                         sizeof(host_string), serv_string,
++                         sizeof(serv_string), NI_NUMERICHOST | NI_NUMERICSERV);
++
+     /* Lookup address from table */
+     addr_key = (ucs_sock_addr_t){
+         .addr = (const struct sockaddr *) &conn_request_attrs.client_address,
+         .addrlen = sizeof(conn_request_attrs.client_address)};
+     na_ucx_addr = na_ucx_addr_map_lookup(&na_ucx_class->addr_map, &addr_key);
+-    NA_CHECK_SUBSYS_ERROR_NORET(addr, na_ucx_addr != NULL, error,
+-        "An entry is already present for this address");
++ 
++
++    if (na_ucx_addr != NULL) {
++        NA_LOG_SUBSYS_WARNING(addr,
++	    "An entry is already present for this address: %s:%s, refcnt: %d",
++            host_string, serv_string, na_ucx_addr->refcount);
++        na_ucx_addr_map_remove(&na_ucx_class->addr_map, &addr_key);
++    }
+ 
+     /* Insert new entry and create new address */
+     na_ret = na_ucx_addr_map_insert(na_ucx_class, &na_ucx_class->addr_map,
+         &addr_key, conn_request, &na_ucx_addr);
++
+     NA_CHECK_SUBSYS_NA_ERROR(
+         addr, error, na_ret, "Could not insert new address");
+ 
```

Comment:
we should bump ref count by 1 here, to account for entry being used by new client. eventual disconnect for previous client will arrive, dropping the ref count. Without this, we end up deleting this newly created entry when previous clients disconnect even finally arrives.